### DISCUSSION TITLE
Is there a way to configure the provider staleTime based on URL?

### DISCUSSION THREAD

**Discussion by mwmcode** [TanStack/query#1293] [Ideas] [answered]
Hello,

Is there a way to make RQ dynamically set `staleTime` for requests that match a certain regex? 

For example, it will automatically set `staleTime: Infinity` for requests that begin with `/api/reference/*` 

Cheers

**Comment by TkDodo** [comment, marked as answer]
I don't think so, but you could get away with a custom hook, something like that:

```
const queryApiReference = (url) => axios.get(`/api/reference/${url}`)

const useApiReferenceQuery = (key, url, options) =>
    useQuery(['apiReference', ...key], () => queryApiReference(url), { ...options, staleTime: Infinity }
```

you could also pass the whole url and set a different `staleTime` depending on `url.startsWith` to make it even more generic or so ...

**Reply by mwmcode** [reply]
I thought a custom hook could be the answer. This is a nice one, thanks.
